### ISSUE TITLE
Allow adding arbitrary anchors to be linked with `@ref`, e.g. via `[content](@id anchor-name)` syntax

### ISSUE THREAD

**Issue by anthony-wang** [JuliaDocs/Documenter.jl#745]
My use case is that I generated thumbnails as previews for plot figures, and I'd like that these thumbnail images be cross-referenced with `@ref` or similar so that when clicked, we can jump to another section in the docs (e.g. to view the full-sized plot or the source code).

Currently, `![alt text](figure.png)` inserts a figure and `[Some-Header-Title](@ref)` inserts a parsed cross-reference HTML link to the specified header anchor, and `[text](URL)` is a hyperlink, but I haven't been able to figure out a way to combine inserting the fig + a cross-referenced link. As far as I can tell, `Documenter` only allows `@ref` to headers.

Therefore, a way of specifying arbitrary anchors on any content so that we can `@ref` them elsewhere in the documentation would be nice!

A suggested design could be:
`[content](@anchor anchor-name)` in the markdown, which is then outputted with `<a>` tags with `id`'s in the HTML. Then these anchors can be referenced by using `@ref`.

(thanks to @mortenpi for the suggested design!)

**Comment by ettersi** [comment]
+1 for this, but shouldn't the syntax be `[content](@id anchor-name)` to be consistent with the [one for naming headers](https://juliadocs.github.io/Documenter.jl/stable/man/syntax/#Duplicate-Headers)?

**Comment by mortenpi** [comment]
Yeah, re-using at-id makes sense to me. Currently, we're being extremely conservative about where we try to find at-ids (namely, only only within top-level headings), so generalizing it shouldn't be a problem.

I wonder how easy it is to introduce arbitrary anchors into LaTeX though. @odow, you don't happen to have any thoughts on this?

**Comment by ettersi** [comment]
I believe this should do the job?
https://www.overleaf.com/learn/latex/Hyperlinks#Inserting_links_manually

**Comment by odow** [comment]
We already use a mix of hyper refs:

https://github.com/JuliaDocs/Documenter.jl/blob/998d26e568a20e3b1764d0c2cef30ebe4d2563ae/src/Writers/LaTeXWriter.jl#L317
https://github.com/JuliaDocs/Documenter.jl/blob/998d26e568a20e3b1764d0c2cef30ebe4d2563ae/src/Writers/LaTeXWriter.jl#L356

But yeah, some way of labeling and ref-ing to tables and figures would be useful.

**Comment by fingolfin** [comment]
See also issue #2584 for a related feature request.

**Comment by moble** [comment]
This would be a really nice feature to have.  It would be extra helpful if these anchors could be added to the `objects.inv` file, so downstream packages could `@extref` them via `DocumenterInterLinks`.  That might get us closer to the dream of actually referencing specific equations in our docs, too!

**Comment by goerz** [comment]
> It would be extra helpful if these anchors could be added to the objects.inv file

As a general rule, I’d like for everything in a Documenter-based documentation that is a link target to also be listed in the inventory, so I would strongly encourage any PR to include that functionality

**Comment by kahliburke** [comment]
Opened #2948 (draft) implementing this — `[content](@id name)` now attaches a named, cross-referenceable anchor to arbitrary inline content (text, images), reusing the existing `@id`/`@ref` machinery as suggested above. It resolves across pages, shares one id namespace with header labels, and (per @moble / @goerz) writes the anchors into `objects.inv` as `std:label` entries so they are reachable via `DocumenterInterLinks` `@extref`. Feedback welcome.
